packages/trader/src/AppV2/Components/TradeParameters/trade-parameters.tsx

```diff
@@ -29,28 +31,40 @@ const TradeParameters = observer(({ is_minimized }: TTradeParametersProps) => {
     };
 
     return (
-        <div
-            className={clsx(
-                'trade-params__options__wrapper',
-                is_minimized && 'trade-params__options__wrapper--minimized'
-            )}
-        >
-            {isVisible('trade_type_tabs') && <TradeTypeTabs is_minimized={is_minimized} />}
-            {isVisible('last_digit') && <LastDigitPrediction is_minimized={is_minimized} />}
-            {isVisible('duration') && <Duration is_minimized={is_minimized} />}
-            {isVisible('strike') && <Strike is_minimized={is_minimized} />}
-            {isVisible('payout_per_point') && <PayoutPerPoint is_minimized={is_minimized} />}
-            {isVisible('barrier') && <Barrier is_minimized={is_minimized} />}
-            {isVisible('growth_rate') && <GrowthRate is_minimized={is_minimized} />}
-            {isVisible('multiplier') && <Multiplier is_minimized={is_minimized} />}
-            {isVisible('stake') && <Stake is_minimized={is_minimized} />}
-            {isVisible('allow_equals') && <AllowEquals is_minimized={is_minimized} />}
-            {isVisible('take_profit') && <TakeProfit is_minimized={is_minimized} />}
-            {isVisible('risk_management') && <RiskManagement is_minimized={is_minimized} />}
-            {/* {isVisible('expiration') && <MultipliersExpirationInfo />} */}
-            {isVisible('accu_info_display') && <AccumulatorsInformation is_minimized={is_minimized} />}
-            {isVisible('mult_info_display') && <MultipliersInformation is_minimized={is_minimized} />}
-        </div>
+        <React.Fragment>
+            <div
+                className={clsx(
+                    'trade-params__options__wrapper',
+                    is_minimized && 'trade-params__options__wrapper--minimized'
+                )}
+            >
+                {isVisible('trade_type_tabs') && <TradeTypeTabs is_minimized={is_minimized} />}
```

Comment:
@henry-deriv , while looking into this code, I felt there is a possibility of improvement. isVisible('someText') calls getTradeParams(symbol)[contract_type].includes looks we loop an array. So for every call isVisible once it loops array. What about we write a function in which we reduce getTradeParams(symbol)[contract_type] array to object. so very time isVisible is called it will just look for key in object, so we save on loop(includes). see if possible or I missed something

Reply:
@vinu-deriv Those arrays' length is too small to cause performance issues.
As it's stated in [an article](https://dev.to/danielbayerlein/array-prototype-includes-can-slow-down-your-code-43c7) I found that suggests replacing **includes** with Set's **has**: '`It makes no sense to replace Array.prototype.includes() with Set.prototype.has() everywhere, because it is not always faster.` '  :)